lib/model/model.go

```diff
@@ -1945,13 +1935,39 @@ func (m *Model) internalScanFolderSubdirs(ctx context.Context, folder string, su
 
 	runner.setState(FolderScanning)
 
-	fchan, err := scanner.Walk(ctx, scanner.Config{
+	subPrefixes := subDirs
+	if len(subPrefixes) == 0 {
+		subPrefixes = []string{""}
+	}
+	haveChan := make(chan *protocol.FileInfo)
+	dbCtx, dbCancel := context.WithCancel(ctx)
+	go func() {
+		for _, sub := range subPrefixes {
+			select {
+			case <-dbCtx.Done():
```

Comment:
wonder if evaluating Done() once doesn't make more sense.

Reply:
I would assume that this is the same after compiling, but yeah, why not.

Reply:
I don't think it is, as the compiler can't know that done doesn't have side-effects, perhaps it increments some internal counter, in which case it can't evaluate once.